TDDProject/Controllers/UserController.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using TDDProject.Models;
+
+namespace TDDProject.Controllers
+{
+    [Route("api/user")]
+    [ApiController]
+    public class UserController : ControllerBase
+    {
+        [HttpGet]
+        [Route("getById/{userId}")]
+        public async Task<IActionResult> GetByIdAsync(int userId)
+        {
+            if (userId == 0)
+            {
+                return BadRequest();
+            }
+            var users = new List<User>()
+            {
+                new User()
+                {
+                    UserId = 1,
+                    Name = "Test",
+                    Address = "Pune"
+                },
+                new User()
+                {
+                    UserId = 2,
+                    Name  ="Test 2",
+                    Address = "Chennai"
+                }
+            };
+            var user = users.Where(x => x.UserId == userId).FirstOrDefault();
+            if (user != null)
+            {
+                return Ok(user);
+            }
+            return NoContent();
```

Comment:
getby id should return 404 when user is not found

Reply:
I suggested to have nocontent instead of notfound as nocontent  will be an success response